### PR TITLE
[pylint] re-enable consider-using-{with, f-string}

### PR DESCRIPTION
**Pull request by pponnuvel**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [x] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: plugins_overview.py [hunk 1]
```diff
@@ -90,23 +89,25 @@ def add_all_items(method, dest, wrapopen=r'\(', wrapclose=r'\)'):
...
-    add_all_items("add_forbidden_path", plugs_data[plugname]['forbidden'])
-    add_all_items("add_cmd_output", plugs_data[plugname]['commands'])
-    add_all_items("collect_cmd_output", plugs_data[plugname]['commands'])
-    add_all_items("add_service_status", plugs_data[plugname]['service_status'])
-    add_all_items("add_journal", plugs_data[plugname]['journals'])
-    add_all_items("add_env_var", plugs_data[plugname]['env'])
+    with open(os.path.join(PLUGDIR, plugfile)).read().replace('\n', '') as pfd:
+        add_all_items(
+            "from sos.report.plugins import ", plugs_data[plugname]['distros'],
+            pfd, wrapopen='', wrapclose='(class|from|import)'
```
Comment (pmoravec) [inline plugins_overview.py:95]: The "pfd" is an extra argument here, compared to current code - is that correct? (I need to understand that by myself as well)

File: tests/cleaner_tests/existing_archive.py [hunk 2]
```diff
@@ -96,8 +96,8 @@ def test_no_empty_obfuscations(self):
             map_json = json.load(mf)
         for mapping in map_json:
             for key, val in map_json[mapping].items():
-                assert key, "Empty key found in %s" % mapping
-                assert val, "%s mapping for '%s' empty" % (mapping, key)
+                assert key, f"Empty key found in {ARCHIVE}"
```
Comment (pmoravec) [inline tests/cleaner_tests/existing_archive.py:99]: Shouldn't it be `{mapping}` instead of `{ARCHIVE}`? Or why the change?

File: sos/cleaner/archives/__init__.py [hunk 3]
```diff
@@ -23,17 +23,27 @@
...
+    with tarfile.open(archive_path) as archive:
+        path = os.path.join(tmpdir, 'cleaner')
+        # set extract filter since python 3.12 (see PEP-706 for more)
+        # Because python 3.10 and 3.11 raises false alarms as exceptions
+        # (see #3330 for examples), we can't use data filter but must
+        # fully trust the archive (legacy behaviour)
+        archive.extraction_filter = getattr(tarfile, 'fully_trusted_filter',
+                                            (lambda member, path: member))
+
+        # Guard against "Arbitrary file write during tarfile extraction"
```
[the remark this answers is not in the document]
Comment (pmoravec) [reply sos/cleaner/archives/__init__.py:35]: Interesting: I was afraid of negative runtime impact of this change so I run cleaner on some sosreport tarball twice: once without the patch and once with the patch. The difference was statistically insignificant (a fraction of a second from 3m cleaner run).

File: tests/cleaner_tests/existing_archive.py (hunk 2 above)
Comment (pponnuvel) [reply tests/cleaner_tests/existing_archive.py]: That was simply a mistake :) Fixed now.

File: sos/cleaner/archives/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pponnuvel) [reply sos/cleaner/archives/__init__.py:35]: Thanks. I think that's a reasonable. 

If we want to fix "Arbitrary file write during tarfile extraction", I think this is the only way (check each file and confirm). Alternative is to ignore it altogether  (i.e. keep this code as before without this change) if we think the source tar file is always trustable.

[the remark this answers is not in the document]
Comment (pponnuvel) [reply sos/cleaner/archives/__init__.py:35]: Looking at https://github.com/python/cpython/blob/main/Lib/tarfile.py#L2265, `extract()` and `extractall()` pretty much work in the same way. 

The major difference between previous and the change in this PR is one additional syscall (`getcwd`) due to `os.path.abspath` call. Cost of a single `getcwd` call varies depending on few factors. 

A quick test on my system:
```
ponnuvel@magicbox:~$ cat t.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <time.h>

int main() {
    struct timespec start, end;
    char cwd[1024];

    if (clock_gettime(CLOCK_MONOTONIC, &start) == -1) {
        perror("clock_gettime");
        exit(0);
    }

    if (getcwd(cwd, sizeof(cwd)) == NULL) {
        perror("getcwd");
        exit(0);
    }

    if (clock_gettime(CLOCK_MONOTONIC, &end) == -1) {
        perror("clock_gettime");
        exit(0);
    }

    // Calculate the elapsed time in nanoseconds
    long seconds = end.tv_sec - start.tv_sec;
    long nanoseconds = end.tv_nsec - start.tv_nsec;
    long elapsed = seconds * 1000000000L + nanoseconds;

    printf("Current working directory: %s\n", cwd);
    printf("Execution time of getcwd: %ld nanoseconds\n", elapsed);
}
ponnuvel@magicbox:~$ gcc -O3 t.c
ponnuvel@magicbox:~$ ./a.out
ponnuvel@magicbox:~$ pwd 
/home/ponnuvel
ponnuvel@magicbox:~$ ./a.out 
Current working directory: /home/ponnuvel
Execution time of getcwd: 4512 nanoseconds
ponnuvel@magicbox:~$ mkdir -p a/b/c/d/e/f/g/h/i/j/k
ponnuvel@magicbox:~$ cp a.out a/b/c/d/e/f/g/h/i/j/k/
ponnuvel@magicbox:~$ cd -
/home/ponnuvel/a/b/c/d/e/f/g/h/i/j/k
ponnuvel@magicbox:~/a/b/c/d/e/f/g/h/i/j/k$ pwd
/home/ponnuvel/a/b/c/d/e/f/g/h/i/j/k
ponnuvel@magicbox:~/a/b/c/d/e/f/g/h/i/j/k$ ./a.out 
Current working directory: /home/ponnuvel/a/b/c/d/e/f/g/h/i/j/k
Execution time of getcwd: 5415 nanoseconds
```
(a relatively low-spec system: 4CPU, i7-6500U CPU @ 2.50GHz, 12G memory)

So it takes about 4.5ms to 5.5ms on my system.

Let's round it up to 6ms per call. On a large sosreport I've (1.5GB), it contains ~180K file:
```
$ tar -tf sosreport-mon2-00384261-2024-07-12-czwswhw.tar.xz | wc -l
179703
```

6ms (per getcwd call) * 180k files ~= 1.08 seconds extra time on my system. I'd think most servers where sosreports are manipulated/extracted are more powerful than my 2016 laptop (where I ran this test). So the extra cost would liekly be less than 1 second for a reasonably sized sosreport.

[the remark this answers is not in the document]
Comment (pponnuvel) [reply sos/cleaner/archives/__init__.py:35]: I should add: please don't think I am trying to justify the change. I am simply too curious and woken up early on a Sunday morning ;-)

I am totally fine to revert this change if the conclusion is "we can always trust the source tar files here".

**Review by pmoravec — APPROVED** [review]

**Review by arif-ali — APPROVED** [review]

**Review by TurboTurtle — APPROVED** [review]
